File: System/Plugins.cs
using System;
using System.Collections.Generic;

// File: Psychosis/Gameplay/System/ObservationPluginSystem.cs

public class Observation
{
    private List<object> plugins;

    public Observation()
    {
        plugins = new List<object>();
    }

    public void Record(string observationName, object properties, object participant)
    {
        // Record the observation
        // "Encyclopedia" is an existing object where observations are stored
        Encyclopedia[observationName] = properties;
    }

    public void AddPlugin(object plugin)
    {
        // Add plugin to the list
        plugins.Add(plugin);
    }

    public void TriggerPlugins(string eventName, object data)
    {
        // Trigger plugins
        foreach (var plugin in plugins)
        {
            try
            {
                var method = plugin.GetType().GetMethod(eventName);
                if (method != null)
                {
                    method.Invoke(plugin, new object[] { data });
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error executing plugin for event '{eventName}': {ex.Message}");
            }
        }
    }
}

// Example usage of plugin system
public class CustomPlugin
{
    public void Record(object data)
    {
        // Implement custom observation recording logic here
        Console.WriteLine("Custom plugin: New observation recorded " + data.ToString());
    }
}

class Program
{
    static void Main(string[] args)
    {
        // Create an instance of the Observation class
        Observation observation = new Observation();

        // Add the custom plugin to the Observation system
        CustomPlugin customPlugin = new CustomPlugin();
        observation.AddPlugin(customPlugin);

        // Trigger custom observation event
        object eventData = new object(); // Assuming eventData is defined somewhere
        observation.TriggerPlugins("CUSTOM_OBSERVATION", eventData);
    }
}

File: System/Plugins.py
# File: Psychosis/Gameplay/System/ObservationPluginSystem.py

class Observation:
    def __init__(self):
        self.plugins = []

    def record(self, observation_name, properties, participant):
        # Record the observation
        # "Encyclopedia" is an existing object where observations are stored
        Encyclopedia[observation_name] = properties

    def add_plugin(self, plugin):
        # Add plugin to the list
        self.plugins.append(plugin)

    def trigger_plugins(self, event, data):
        # Trigger plugins
        for plugin in self.plugins:
            try:
                if hasattr(plugin, event):
                    getattr(plugin, event)(data)
            except Exception as e:
                print(f"Error executing plugin for event '{event}': {e}")

# Example event naming convention
EVENTS = {
    'CUSTOM_OBSERVATION': 'custom_observation'
}

# Example usage of plugin system
class CustomPlugin:
    def record(self, data):
        # Implement custom observation recording logic here
        print("Custom plugin: New observation recorded", data)

# Create an instance of the Observation class
observation = Observation()

# Add the custom plugin to the Observation system
custom_plugin = CustomPlugin()
observation.add_plugin(custom_plugin)

# Trigger custom observation event
event_data = {}  # Assuming eventData is defined somewhere
observation.trigger_plugins(EVENTS['CUSTOM_OBSERVATION'], event_data)
